src/utils/workspace.ts

```diff
@@ -31,32 +32,36 @@ export async function selectWorkspaceFile(placeHolder: string, getSubPath?: (f:
         getSubPath);
 }
 
-export async function selectWorkspaceItem(placeHolder: string, options: vscode.OpenDialogOptions, getSubPath?: (f: vscode.WorkspaceFolder) => string | undefined): Promise<string> {
+export async function selectWorkspaceItem(placeHolder: string, options: vscode.OpenDialogOptions, getSubPath?: (f: vscode.WorkspaceFolder) => string | undefined, fileExtension?: string): Promise<string> {
     let folder: IAzureQuickPickItem<string | undefined> | undefined;
+    let quickPicks: IAzureQuickPickItem<string | undefined>[] = [];
     if (vscode.workspace.workspaceFolders) {
-        const folderPicks: IAzureQuickPickItem<string | undefined>[] = vscode.workspace.workspaceFolders.map((f: vscode.WorkspaceFolder) => {
-            let subpath: string | undefined;
-            if (getSubPath) {
-                subpath = getSubPath(f);
-            }
+        // if there's a fileExtension, then only populate the quickPick menu with that, otherwise show the current folders in the workspace
+        quickPicks = fileExtension ? await findFilesByFileExtension(undefined, fileExtension) :
+            vscode.workspace.workspaceFolders.map((f: vscode.WorkspaceFolder) => {
+                let subpath: string | undefined;
+                if (getSubPath) {
+                    subpath = getSubPath(f);
+                }
 
-            const fsPath: string = subpath ? path.join(f.uri.fsPath, subpath) : f.uri.fsPath;
-            return { label: path.basename(fsPath), description: fsPath, data: fsPath };
-        });
+                const fsPath: string = subpath ? path.join(f.uri.fsPath, subpath) : f.uri.fsPath;
+                return { label: path.basename(fsPath), description: fsPath, data: fsPath };
+            });
 
-        folderPicks.push({ label: '$(file-directory) Browse...', description: '', data: undefined });
-        folder = await ext.ui.showQuickPick(folderPicks, { placeHolder });
+        quickPicks.push({ label: '$(file-directory) Browse...', description: '', data: undefined });
+        folder = await ext.ui.showQuickPick(quickPicks, { placeHolder });
     }
 
     return folder && folder.data ? folder.data : (await ext.ui.showOpenDialog(options))[0].fsPath;
 }
 
-export async function showWorkspaceFoldersQuickPick(placeHolderString: string, context: IActionContext, subPathSetting: string | undefined): Promise<string> {
-    context.telemetry.properties.cancelStep = 'showWorkspaceFolders';
+export async function showWorkspaceFolders(placeHolderString: string, context: IActionContext, subPathSetting: string | undefined, fileExtension?: string): Promise<string> {
+    context.telemetry.properties.cancelStep = 'showWorkspaceFoldersAndExtensions';
     return await selectWorkspaceItem(
         placeHolderString,
         {
-            canSelectFiles: false,
+            // on Windows, if both files and folder are set to true, then it defaults to folders
+            canSelectFiles: true,
```

Comment:
> I'm wondering if I should just filter out for .zip/.jar/.war files by default since I think that's all we support but wasn't sure if we'd want to have arbitrary restrictions like that.

Yes I think you should only show those file types